src/ripple/overlay/impl/Handshake.cpp

```diff
@@ -34,6 +34,73 @@
 
 namespace ripple {
 
+std::optional<std::string>
+getFeatureValue(
+    boost::beast::http::fields const& headers,
+    std::string const& feature)
+{
+    auto const header = headers.find("X-Protocol-Ctl");
+    if (header == headers.end())
+        return {};
+    boost::smatch match;
+    boost::regex rx(feature + "=([^;\\s]+)");
+    auto const value = header->value().to_string();
+    if (boost::regex_search(value, match, rx))
+        return {match[1]};
+    return {};
+}
+
+bool
+isFeatureValue(
+    boost::beast::http::fields const& headers,
+    std::string const& feature,
+    std::string const& value)
+{
+    auto const fvalue = getFeatureValue(headers, feature);
+    if (!fvalue)
+        return false;
+    auto test = [&](auto r) {
+        boost::regex rx(r);
+        return boost::regex_search(fvalue.value(), rx);
+    };
+    // could be a single value or a comma separated list of values
+    return test("^" + value + "$") || test("^" + value + ",") ||
+        test("," + value + ",") + test("," + value + "$");
```

Comment:
That works, but it ends up searching the string four times, doing basically the same search every time. We could do a couple of things here:
1. Tokenize `fvalue`, using comma as a delimiter, and check the token for equality with value. We'd have to write our own tokenizer (strtok requires a non-const `char*`), but that should be straight-forward.
2. Use a more complex regular expression, where the first char can either be a comma or the start, and the last char can either be a comma or the end.
3. Do a single search and check the boundaries.

But I don't think we need to do any of that. There's already a function in the code-base that's perfect for this: `beast::rfc2616::split_commas`. It even correctly handles cases where commas appear in quotes.

Reply:
Thanks, updated.

Reply:
I know I suggested `split_commas`, but take a look at rfc2616.h and see if `token_in_list` would be better, as that does not allocate.

Reply:
Changed.